src/compiler/types.ts

```diff
@@ -1110,11 +1109,12 @@ namespace ts {
     export interface QualifiedName extends Node {
         readonly kind: SyntaxKind.QualifiedName;
         readonly left: EntityName;
-        readonly right: Identifier;
-        /*@internal*/ jsdocDotPos?: number;                      // QualifiedName occurs in JSDoc-style generic: Id1.Id2.<T>
+        readonly right: Identifier | StringLiteralLike;
     }
 
-    export type EntityName = Identifier | QualifiedName;
+    export type EntityName = Identifier | QualifiedName | StringLiteralLike;

```

Comment:
If the only kind of EntityName that can start with a StringLiteralLike is the qualifier on an ImportTypeNode, would it make more sense to make that a special case and leave the meaning of `EntityName` unchanged?

Reply:
Maybe; but when going in that direction, the suite of utility functions that needed to handle it was less clear. It could be worth the change, regardless.